svtyper/singlesample.py

```diff
@@ -0,0 +1,851 @@
+    over_threshold = False
+    (regionA, regionB) = regions
+    (countA, countB) = ( count_reads_in_region(regionA, bam), count_reads_in_region(regionB, bam) )
+    if countA > max_reads or countB > max_reads:
+        over_threshold = True
+        msg = ("SKIPPING -- Variant '{}' has a region with too many reads (> {})\n"
+                "\t\t A: (sample={} chrom={} center={} leftflank={} rightflank={}) : {}\n"
+                "\t\t B: (sample={} chrom={} center={} leftflank={} rightflank={}) : {}").format(
+                        variant_id,
+                        max_reads,
+                        regionA[0], regionA[1], regionA[2], regionA[3], regionA[4],
+                        countA,
+                        regionB[0], regionB[1], regionB[2], regionB[3], regionB[4],
+                        countB,
+                )
+        logit(msg)
+        return (over_threshold, [])
+
+    reads_generator = chain(
+        get_reads_iterator(regionA, bam),
+        get_reads_iterator(regionB, bam)
+    )
+    return (over_threshold, reads_generator)
+
+def gather_reads(bam, variant_id, regions, library_data, active_libs, max_reads):
+    fragment_dict = {}
+    (over_threshold, reads) = retrieve_reads_from_db(bam, variant_id, regions, max_reads)
+
+    for read in reads:
+        if read.is_unmapped or read.is_duplicate: continue
+        lib = library_data[read.get_tag('RG')]
+        if lib.name not in active_libs: continue
+        if read.query_name in fragment_dict:
+            fragment_dict[read.query_name].add_read(read)
+        else:
+            lib = library_data[read.get_tag('RG')]
+            fragment_dict[read.query_name] = SamFragment(read, lib)
+    return (fragment_dict, over_threshold)
+
+def blank_genotype_result():
+    return {
+        'qual' : 0,
+        'formats': {
+            'GT'  : './.',
+            'GQ'  : '.',
+            'SQ'  : '.',
+            'GL'  : '.',
+            'DP'  : 0,
+            'AO'  : 0,
+            'RO'  : 0,
+            'AS'  : 0,
+            'ASC' : 0,
+            'RS'  : 0,
+            'AP'  : 0,
+            'RP'  : 0,
+            'QR'  : 0,
+            'QA'  : 0,
+            'AB'  : '.',
+        }
+    }
+
+def make_empty_genotype_result(variant_id, sample_name):
+    return {
+        'variant.id' : variant_id,
+        'sample.name' : sample_name,
+        'genotype' : {}
+    }
+
+def make_detailed_empty_genotype_result(variant_id, sample_name):
+    return {
+        'variant.id' : variant_id,
+        'sample.name' : sample_name,
+        'genotype' : blank_genotype_result()
+    }
+
+
+def gather_split_read_evidence(sam_fragment, breakpoint, split_slop, min_aligned):
+    (ref_seq, alt_seq, alt_clip) = (0, 0, 0)
+
+    elems = ('chrom', 'pos', 'ci', 'is_reverse')
+    (chromA, posA, ciA, o1_is_reverse) = tuple(breakpoint['A'][i] for i in elems)
+    (chromB, posB, ciB, o2_is_reverse) = tuple(breakpoint['B'][i] for i in elems)
```

Comment:
I understand this is trivial and I don't think it needs to change, but I hadn't seen tuple used in this way before and was curious if there was an impact versus a list comprehension. Surprisingly, it seems there is a difference and I wanted to share.

```
>>> timeit.timeit("(u, v, w) = tuple(x['A'][i] for i in ('a', 'b', 'c'))", setup="x = { 'A': { 'a': 1, 'b': 66, 'c': 77 } }", number=1000000)
1.2717030048370361
>>> timeit.timeit("(u, v, w) = (x['A'][i] for i in ('a', 'b', 'c'))", setup="x = { 'A': { 'a': 1, 'b': 66, 'c': 77 } }", number=1000000)
0.6487770080566406
>>> timeit.timeit("(u, v, w) = [x['A'][i] for i in ('a', 'b', 'c')]", setup="x = { 'A': { 'a': 1, 'b': 66, 'c': 77 } }", number=1000000)
0.3766791820526123
```